sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -112,49 +122,65 @@ static void otbn_write(uint32_t dest_addr, const uint32_t *src,
 static rom_error_t otbn_cmd_run(otbn_cmd_t cmd, rom_error_t error) {
   enum {
     kIntrStateDone = (1 << OTBN_INTR_COMMON_DONE_BIT),
-    // Use a bit index that doesn't overlap with `otbn_err_bits_t`.
+    // Use a bit index that doesn't overlap with error bits.
     kResDoneBit = 31,
   };
   static_assert((UINT32_C(1) << kResDoneBit) > kOtbnErrBitsLast,
-                "kResDoneBit must not overlap with `otbn_err_bits_t`");
+                "kResDoneBit must not overlap with OTBN error bits");
 
   abs_mmio_write32(kBase + OTBN_INTR_STATE_REG_OFFSET, kIntrStateDone);
   abs_mmio_write32(kBase + OTBN_CMD_REG_OFFSET, cmd);
 
   rom_error_t res = kErrorOk ^ (UINT32_C(1) << kResDoneBit);
   uint32_t reg = 0;
-  while (launder32(reg) != kIntrStateDone) {
+  do {
     reg = abs_mmio_read32(kBase + OTBN_INTR_STATE_REG_OFFSET);
     res ^= bitfield_bit32_read(reg, OTBN_INTR_COMMON_DONE_BIT) << kResDoneBit;
-  }
+  } while (launder32(reg) != kIntrStateDone);
   HARDENED_CHECK_EQ(reg, kIntrStateDone);
   abs_mmio_write32(kBase + OTBN_INTR_STATE_REG_OFFSET, kIntrStateDone);
 
-  otbn_err_bits_t err_bits;
-  otbn_err_bits_get(&err_bits);
+  // Error bits register should be 0 (no errors).
+  uint32_t err_bits = abs_mmio_read32(kBase + OTBN_ERR_BITS_REG_OFFSET);
   res ^= err_bits;
 
   // Status should be kOtbnStatusIdle; OTBN can also issue a done interrupt
   // when transitioning to the "locked" state, so it is important to check
   // the status here.
   uint32_t status = abs_mmio_read32(kBase + OTBN_STATUS_REG_OFFSET);
 
-  if (launder32(res) == kErrorOk && launder32(status) == kOtbnStatusIdle) {
+  if (launder32(res) == kErrorOk && launder32(err_bits) == 0 &&
+      launder32(status) == kOtbnStatusIdle) {
     HARDENED_CHECK_EQ(res, kErrorOk);
+    HARDENED_CHECK_EQ(err_bits, 0);
     HARDENED_CHECK_EQ(abs_mmio_read32(kBase + OTBN_STATUS_REG_OFFSET),
                       kOtbnStatusIdle);
     return res;
   }
   return error;
 }
 
+static inline void otbn_set_ctrl_software_errs_fatal(bool enable) {
```

Comment:
I can't seem to find a call site with `enable = false`. Can you inline this like
`sec_mmio_write32(kBase + OTBN_CTRL_REG_OFFSET, (1 << OTBN_CTRL_SOFTWARE_ERRS_FATAL_BIT));`